Formula/tcl-tk.rb

```diff
@@ -98,9 +95,30 @@ def install
                             "--mandir=#{man}"
       system "make", "install"
     end
+
+    resource("itk4").stage do
+      itcl_dir = Pathname.glob(lib/"itcl*").last
+      args = %W[
+        --prefix=#{prefix}
+        --exec-prefix=#{prefix}
+        --with-tcl=#{lib}
+        --with-tk=#{lib}
+        --with-itcl=#{itcl_dir}
+      ]
+      system "./configure", *args
+      system "make"
+      system "make", "install"
+    end
   end
 
   test do
     assert_equal "honk", pipe_output("#{bin}/tclsh", "puts honk\n").chomp
+
+    # Check Itk pkg is found, loads the .dylib, and reports its version
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
So... itk is an object oriented GUI framework, and TBH, I'm not really sure how you could go about testing it from the command line. The test I added isn't simply a version test: it's used to check that the library successfully loads and initializes: this verifies that it was compiled correctly, has been installed in the right place and is on TCL's search path.

I'll note that other libraries included in this formula (such as tcltls, critcl etc) don't have any tests at all, so if you'd prefer it, I'd be entirely happy to drop this test entirely.

Reply:
> If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.

Reply:
OK. I've added a test...